include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -153,14 +274,17 @@ class alignment_algorithm :
     {
         auto col = this->current_column();
 
-        this->init_origin_cell(*std::ranges::begin(col), cache);
+        this->init_origin_cell(*seqan3::begin(col), cache);
 
-        ranges::for_each(col | ranges::view::drop_exactly(1), [&cache, this](auto & cell)
+        ranges::for_each(col | ranges::view::drop_exactly(1), [&cache, this](auto && cell)
         {
-            this->init_column_cell(cell, cache);
+            this->init_column_cell(std::forward<decltype(cell)>(cell), cache);
```

Comment:
Yet, always, as it is a zip view internally returning tuple of tuples.

Reply:
A tuple of tuples isn't necessarily heap-allocated. In the case where it contains only builtins, it is not. And since this is "hot code" I am pretty sure copies would be faster, just saying...

Reply:
I don't get this. Dereferencing the zip iterator will return a rvalue tuple. Now why would a copy be better than a perferctly forwarded rvalue? And if the compiler sees that eveything is a builtin he copies anyway.

Reply:
> Now why would a copy be better than a perferctly forwarded rvalue?
> And if the compiler sees that eveything is a builtin he copies anyway.

Yes and no. I think it's still a difference whether you take by copy or take by &&. It will affect the compilers ability to predict aliasing or not?

Anyway, not so important...

Reply:
I think the relevant performance difference (if any) can be evaluated with the simd code